### DISCUSSION TITLE
Getting blocked by people.apache.org when fetching pom files - Resulting in extremely slow scanning

### DISCUSSION THREAD

**Discussion by SemProvoost** [aquasecurity/trivy#8000] [Q&A] [answered]
### Description

When Trivy tries to fetch pom files from people.apache.org, the domain throttles/blocks you, which results in constant timeouts and extremely slow scans.
These people.apache.org are always just 404 or not relevant.

### Desired Behavior

Skip the people.apache.org urls instead of trying to fetch.

Example that fixes it:
![image](https://github.com/user-attachments/assets/bd9c3229-39aa-4126-8ebb-0e328c06d163)

### Actual Behavior

Tries fetching from an unreliable source which makes scans very slow (without any benefits)

![image](https://github.com/user-attachments/assets/f758c06d-7e05-4d83-aa7e-8dc6c76830b7)

### Reproduction Steps

```bash
Having a bit of a more complicated repo with multiple pom files where Trivy tries to fetch from people.apache.org
```

### Target

Git Repository

### Scanner

Vulnerability

### Output Format

None

### Mode

Standalone

### Debug Output

```bash
screenshot in 'Actual Behaviour' section
```

### Operating System

verified on macOS & Linux

### Version

```bash
0.56.1
```

### Checklist

- [X] Run `trivy clean --all`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
Please share a small pom.xml to reproduce it. We need to figure out why it accesses people.apache.org first.

**Reply by SemProvoost** [reply]
Here's a small sample to reproduce:

**root/pom.xml**
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven.xsd">

    <modelVersion>4.0.0</modelVersion>
    <groupId>com.aaa</groupId>
    <artifactId>bbb</artifactId>
    <version>1.0</version>
    <packaging>pom</packaging>
    <name>Parent</name>

    <dependencyManagement>
        <dependencies>

            <dependency>
                <groupId>com.amazonaws</groupId>
                <artifactId>amazon-sqs-java-messaging-lib</artifactId>
                <version>1.0.4</version>
            </dependency>

        </dependencies>
    </dependencyManagement>

</project>
```

**root/c/pom.xml**
```
<project xmlns="http://maven.apache.org/POM/4.0.0"
		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">

	<parent>
		<artifactId>bbb</artifactId>
		<groupId>com.aaa</groupId>
		<version>1.0</version>
		<relativePath>../pom.xml</relativePath>
	</parent>

	<modelVersion>4.0.0</modelVersion>
	<artifactId>shared</artifactId>
	<packaging>jar</packaging>
	<name>Shared Module</name>

	<dependencies>

		<dependency>
			<groupId>com.amazonaws</groupId>
			<artifactId>amazon-sqs-java-messaging-lib</artifactId>
		</dependency>

	</dependencies>

</project>
```


command:
`trivy fs . --scanners vuln --debug`

You might need to run it a few times until people.apache.org is throttling/blocking your IP.

**Reply by knqyf263** [reply]
@DmitriyLewen Can you please take a look?

**Reply by SemProvoost** [reply]
There's a PR open for this [here](https://github.com/aquasecurity/trivy/pull/8001) btw 😄

**Reply by knqyf263** [reply]
Yes, but that PR is a workaround. We should find a solution if possible.

**Reply by DmitriyLewen** [reply]
Hi @SemProvoost
I can't reproduce your case.

Trivy doesn't access `people.apache.org` for these `pom.xml` files.

We collect `repositories` from the upper pom files. Perhaps one of your pom.xml files uses this repository?

**Reply by SemProvoost** [reply]
@DmitriyLewen make sure to trigger it multiple times until the people.apache.org starts throttling you.

Trivy only talks about actually trying to connect with 'people.apache.org' in their error messages.

I reproduced this with the exact steps in this thread 😁
![image](https://github.com/user-attachments/assets/c55c746e-6799-4e42-9025-a9a0c25722c4)

**Reply by SemProvoost** [reply]
You could make a few hundred apicalls to `https://people.apache.org/repo/m2-incubating-repository/org/apache/geronimo/genesis/genesis/1.2/genesis-1.2.pom` for example to get yourself throttled, then run trivy on the root folder of the example. Let me know if you run into issues, available for a call as well if needed.

**Reply by DmitriyLewen** [reply, marked as answer]
I got it. You use `v0.56.1` version.
We updated logic to work with parents in `v0.57.0` (#7541)

Can you take a looks with latest version?

**Reply by DmitriyLewen** [reply]
Also I want to add that Trivy finds `people.apache.org` from [org.apache.geronimo.genesis.config:project-config](https://repo1.maven.org/maven2/org/apache/geronimo/genesis/config/project-config/1.2/project-config-1.2.pom) file.
I am not sure that we need to skip these repositories.
Maybe there is a way to contact the developers and ask them why they added these repositories (despite the fact that these artifacts are already in maven central)

**Reply by SemProvoost** [reply]
With the updated logic, it seems to be all good (and a lot faster) now. (doesn't do the calls to `people.apache.org` anymore)
Thanks for the help! Feel free to close this ticket and PR. 😁




> I am not sure that we need to skip these repositories.

Well, it basically just times out 100% of the time once blocked - So it would only be useful the first (few) times you scan (but for bigger repos, it would already block you on first scan even, I think).
So we were basically falling back to offline mode in these cases.

Anyway, now we're not hitting the `people.apache.org` anymore so all good. Thanks for the follow-up here!
